### PR TITLE
Update Hl1 methodology

### LINKED ISSUE #928: Improvements to Homelessness
Related to #906 

We must further investigate the issue in #906 with James and also to have a discussion with the homelessness team to identify what outputs they are extracting from the SLFs. The aim of this is to identify if this methodology is wrong and whether we can improve this for the June update. 

During the discussion with the homelessness team, we should also investigate their needs for the date calculation flags. We currently have flags for 6 months after, 6 months during and 6 months before a HL1 application. The team may benefit for using 12 month calculations instead.

### PR DESCRIPTION
**Pull request by Jennit07**
closes #928 and #906 

opening this as a draft for now as i am testing this to make sure the new variables are picked up by the episode and individual files

### CODE REVIEW

File: R/process_extract_homelessness.R [hunk 1]
```diff
@@ -145,21 +145,16 @@ process_extract_homelessness <- function(
...
-  } else {
-    filtered_data <- data
-  }
+  data <- data %>%
+    dplyr::left_join(
+      completeness_data %>%
+        dplyr::select(sending_local_authority_name, pct_complete_all),
+      by = dplyr::join_by("sending_local_authority_name")
+    ) %>%
+    dplyr::rename(hl1_completeness = pct_complete_all)
```
Comment (Jennit07) [inline R/process_extract_homelessness.R:154]: removed the filtering here

File: R/process_extract_homelessness.R [hunk 2]
```diff
@@ -174,7 +169,8 @@ process_extract_homelessness <- function(
       hl1_application_ref = "application_reference_number",
       hl1_sending_lca = "sending_local_authority_code_9",
       hl1_property_type = "property_type_code",
-      "hl1_reason_ftm"
+      "hl1_reason_ftm",
+      hl1_completeness
```
Comment (Jennit07) [inline R/process_extract_homelessness.R:173]: new variable for comparing to SG completeness

File: R/process_lookup_homelessness.R [hunk 3]
```diff
@@ -101,6 +101,16 @@ add_homelessness_date_flags <- function(data, year, lookup = create_homelessness
...
+    ) %>%
+    dplyr::mutate(
+      hl1_12_months_pre_app = lubridate::rollback(.data$record_keydate1,
+        months(-12),
+        roll_to_first = TRUE
+      ),
+      hl1_12_months_post_app = lubridate::add_with_rollback(.data$record_keydate2,
+        months(12),
+        roll_to_first = TRUE
+      )
```
Comment (Jennit07) [inline R/process_lookup_homelessness.R:113]: new variables for 12 months pre/post application

File: R/process_extract_homelessness.R (hunk 1 above)
Comment (Jennit07) [reply R/process_extract_homelessness.R:154]: we need to add some code here to round the percentage to one decimal place

**Comment by Jennit07** [comment]
I have tested this to the episode file and i am happy it is working as expected. 

To summarise the changes include: 
- Remove filtering if the data is between 90%-105% completeness or 'East Ayrshire'
- Create a new variable with the completeness percentage compared to SG total assessment decisions (rounded to one decimal place)
- Create a new variables hl1_12_months_pre_app and hl1_12_months_post_app. This will be a date variable.

I investigated what variables may need to show up in the individual file. I have concluded that the new variables do not need to be there as we do not have `record_keydate1` and `record_keydate2` available in the individual file. The only homelessness variable we have available is `hl1_in_fy`. As we do not have `record_keydate1` and `record_keydate2` available i dont think we need to include the date variables for pre/post application.

Ready to finalise this PR and have this reviewed in preparation for the June update. Thanks

**Review by lizihao-anu — APPROVED** [review]

**Review by SwiftySalmon — APPROVED** [review]
looks good!

**Pull request merged**